### ISSUE TITLE
(^)/2 missing in bagof/3 setof/3

### ISSUE THREAD

**Issue by UWN** [ichiban/prolog#223]
```
?- bagof(X, _^ (X=1),Xs). 
X = _642836,
Xs = [1]; % expected
?- bagof(X,_^_^ (X=1),Xs).
2022/05/11 13:53:17 error(existence_error(procedure, ^ /2), 'Unknown procedure.'), unexpected
```

The `(^)/2` for indicating existential variables may be nested. That's why there is such a recursive definition in 7.1.1.3. The example `X^Y^f(X,Y,Z)` is found in the notes.

**Comment by UWN** [comment]
Still here

```
ulrich@TU-Wien:/opt/gupu/ichiban-prolog$ go install github.com/ichiban/prolog/cmd/1pl@latest
go: downloading github.com/ichiban/prolog v0.10.1
ulrich@TU-Wien:/opt/gupu/ichiban-prolog$ $(go env GOPATH)/bin/1pl
Top level for ichiban/prolog v0.10.1
This is for testing purposes only!
See https://github.com/ichiban/prolog for more details.
Type Ctrl-C or 'halt.' to exit.
?- bagof(X,_^_^ (X=1),Xs).
2022/07/20 16:36:29 error(existence_error(procedure,^ /2),bagof/3)

**Comment by ichiban** [comment]
Fixed in `v0.10.2`.

```console
$ go install github.com/ichiban/prolog/cmd/1pl@latest
$ $(go env GOPATH)/bin/1pl
Top level for ichiban/prolog v0.10.2
This is for testing purposes only!
See https://github.com/ichiban/prolog for more details.
Type Ctrl-C or 'halt.' to exit.
?- bagof(X, _^ (X=1),Xs).
X = _1078,
Xs = [1];
?- bagof(X,_^_^ (X=1),Xs).
X = _1122,
Xs = [1];
?- 
```
